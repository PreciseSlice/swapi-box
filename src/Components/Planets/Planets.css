.planetContainer {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  overflow: auto;
  flex-wrap: wrap;
  justify-content: space-evenly; }
