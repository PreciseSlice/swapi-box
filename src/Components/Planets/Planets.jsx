import React from 'react';

const Planets = (props) => {
  return (
    <div>
      <h3>I AM A Planets!</h3>
    </div>
  );
};

export default Planets;