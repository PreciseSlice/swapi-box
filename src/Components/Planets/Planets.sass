.planets 
  color: #feda4a