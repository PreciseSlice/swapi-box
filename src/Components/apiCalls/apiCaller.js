export const callFetch = async url => {
  const getData = await fetch(url);
  const jsonData = await getData.json();

  return await jsonData;
};

export const getFilms = async url => {
  try {
    const fetchFilmData = await callFetch(url);

    const title = await fetchFilmData.title;
    const episodeId = await fetchFilmData.episode_id;
    const openingCrawl = await fetchFilmData.opening_crawl;
    const releaseDate = await fetchFilmData.release_date;

    return await { title, episodeId, openingCrawl, releaseDate };
  } catch (er) {
    const error = new Error('getFilms failed to fetch filmData');
    return error;
  }
};

export const getPeople = async url => {
  try {
    const fetchPeopleData = await callFetch(url);

    const peopleMap = fetchPeopleData.results.map(async person => {
      const name = person.name;

      const fetchHomeworldData = await callFetch(person.homeworld);
      const homeworld = fetchHomeworldData.name;
      const population = fetchHomeworldData.population;

      const fetchSpeciesData = await callFetch(person.species);
      const speciesName = fetchSpeciesData.name;

      return { name, homeworld, population, speciesName };
    });
    return Promise.all(peopleMap);
  } catch (er) {
    const error = new Error('getPeople failed to fetch peopleData');
    return error;
  }
};

export const getPlanets = async url => {
  try {
    const fetchPlanetData = await callFetch(url);

    const planetMap = fetchPlanetData.results.map(async planet => {
      const name = planet.name;
      const terrain = planet.terrain;
      const population = planet.population;
      const climate = planet.climate;

      const residentMap = planet.residents.map(async resident => {
        const fetchResidentData = await callFetch(resident);

        return fetchResidentData.name;
      });

      const mapReturn = await Promise.all(residentMap);
      const residents = mapReturn.join(', ');

      return { name, terrain, population, climate, residents };
    });
    return Promise.all(planetMap);
  } catch (er) {
    const error = new Error('getPlaenets failed to fetch plaenetData');
    return error;
  }
};

export const getVehicles = async url => {
  try {
    const fetchVehicleData = await callFetch(url);

    const vehicleMap = fetchVehicleData.results.map(async vehicle => {
      const name = vehicle.name;
      const model = vehicle.model;
      const vehicleClass = vehicle.vehicle_class;
      const passengers = vehicle.passengers;

      return { name, model, vehicleClass, passengers };
    });
    return Promise.all(vehicleMap);
  } catch (er) {
    const error = new Error('getVehicles failed to fetch vehicleData');
    return error;
  }
};
