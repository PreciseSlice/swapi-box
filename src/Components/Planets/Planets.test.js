/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Planets from './Planets';
import { shallow, mount } from 'enzyme';
import { cleanPlanets } from '../../apiCalls/apiMockData';

describe('Planets', () => {
  let wrapper;
  let mockFunction;
  let secondMockFunction;

  beforeEach(() => {
    mockFunction = jest.fn();
    secondMockFunction = jest.fn();
    wrapper = mount(
      <Planets
        setPlanets={mockFunction}
        planetData={cleanPlanets}
        favClickHandler={secondMockFunction}
        favorites={cleanPlanets}
      />
    );
  });

  it('Exist and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should get planet data if it does not have it, by calling getPlanets.', async () => {
    wrapper = shallow(
      <Planets
        setPlanets={mockFunction}
        planetData={[]}
        favClickHandler={secondMockFunction}
        favorites={cleanPlanets}
      />
    );

    await wrapper.instance().componentDidMount();
    expect(mockFunction).toHaveBeenCalled();
  });

  it('renders a card for each object in the array it is passed', () => {
    expect(wrapper.find('.card').length).toEqual(10);
  });
});
