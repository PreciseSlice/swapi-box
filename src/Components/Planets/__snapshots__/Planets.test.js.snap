// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets Exist and matches snapshot 1`] = `
<Planets
  clickHandler={[Function]}
  favorites={
    Array [
      Object {
        "climate": "temperate",
        "name": "Alderaan",
        "population": "2000000000",
        "residents": ", , ",
        "terrain": "grasslands, mountains",
      },
      Object {
        "climate": "temperate, tropical",
        "name": "Yavin IV",
        "population": "1000",
        "residents": "",
        "terrain": "jungle, rainforests",
      },
      Object {
        "climate": "frozen",
        "name": "Hoth",
        "population": "unknown",
        "residents": "",
        "terrain": "tundra, ice caves, mountain ranges",
      },
      Object {
        "climate": "murky",
        "name": "Dagobah",
        "population": "unknown",
        "residents": "",
        "terrain": "swamp, jungles",
      },
      Object {
        "climate": "temperate",
        "name": "Bespin",
        "population": "6000000",
        "residents": "",
        "terrain": "gas giant",
      },
      Object {
        "climate": "temperate",
        "name": "Endor",
        "population": "30000000",
        "residents": "",
        "terrain": "forests, mountains, lakes",
      },
      Object {
        "climate": "temperate",
        "name": "Naboo",
        "population": "4500000000",
        "residents": ", , , , , , , , , , ",
        "terrain": "grassy hills, swamps, forests, mountains",
      },
      Object {
        "climate": "temperate",
        "name": "Coruscant",
        "population": "1000000000000",
        "residents": ", , ",
        "terrain": "cityscape, mountains",
      },
      Object {
        "climate": "temperate",
        "name": "Kamino",
        "population": "1000000000",
        "residents": ", , ",
        "terrain": "ocean",
      },
      Object {
        "climate": "temperate, arid",
        "name": "Geonosis",
        "population": "100000000000",
        "residents": "",
        "terrain": "rock, desert, mountain, barren",
      },
    ]
  }
  planetData={
    Array [
      Object {
        "climate": "temperate",
        "name": "Alderaan",
        "population": "2000000000",
        "residents": ", , ",
        "terrain": "grasslands, mountains",
      },
      Object {
        "climate": "temperate, tropical",
        "name": "Yavin IV",
        "population": "1000",
        "residents": "",
        "terrain": "jungle, rainforests",
      },
      Object {
        "climate": "frozen",
        "name": "Hoth",
        "population": "unknown",
        "residents": "",
        "terrain": "tundra, ice caves, mountain ranges",
      },
      Object {
        "climate": "murky",
        "name": "Dagobah",
        "population": "unknown",
        "residents": "",
        "terrain": "swamp, jungles",
      },
      Object {
        "climate": "temperate",
        "name": "Bespin",
        "population": "6000000",
        "residents": "",
        "terrain": "gas giant",
      },
      Object {
        "climate": "temperate",
        "name": "Endor",
        "population": "30000000",
        "residents": "",
        "terrain": "forests, mountains, lakes",
      },
      Object {
        "climate": "temperate",
        "name": "Naboo",
        "population": "4500000000",
        "residents": ", , , , , , , , , , ",
        "terrain": "grassy hills, swamps, forests, mountains",
      },
      Object {
        "climate": "temperate",
        "name": "Coruscant",
        "population": "1000000000000",
        "residents": ", , ",
        "terrain": "cityscape, mountains",
      },
      Object {
        "climate": "temperate",
        "name": "Kamino",
        "population": "1000000000",
        "residents": ", , ",
        "terrain": "ocean",
      },
      Object {
        "climate": "temperate, arid",
        "name": "Geonosis",
        "population": "100000000000",
        "residents": "",
        "terrain": "rock, desert, mountain, barren",
      },
    ]
  }
  setPlanets={[Function]}
>
  <div
    className="planet-container"
  >
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "temperate",
          "name": "Alderaan",
          "population": "2000000000",
          "residents": ", , ",
          "terrain": "grasslands, mountains",
        }
      }
      highlighted="highlighted"
      key="Alderaan"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Alderaan
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 2000000000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: , , 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: grasslands, mountains
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "temperate, tropical",
          "name": "Yavin IV",
          "population": "1000",
          "residents": "",
          "terrain": "jungle, rainforests",
        }
      }
      highlighted="highlighted"
      key="Yavin IV"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Yavin IV
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate, tropical
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 1000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: jungle, rainforests
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "frozen",
          "name": "Hoth",
          "population": "unknown",
          "residents": "",
          "terrain": "tundra, ice caves, mountain ranges",
        }
      }
      highlighted="highlighted"
      key="Hoth"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Hoth
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: frozen
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: unknown
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: tundra, ice caves, mountain ranges
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "murky",
          "name": "Dagobah",
          "population": "unknown",
          "residents": "",
          "terrain": "swamp, jungles",
        }
      }
      highlighted="highlighted"
      key="Dagobah"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Dagobah
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: murky
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: unknown
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: swamp, jungles
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "temperate",
          "name": "Bespin",
          "population": "6000000",
          "residents": "",
          "terrain": "gas giant",
        }
      }
      highlighted="highlighted"
      key="Bespin"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Bespin
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 6000000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: gas giant
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "temperate",
          "name": "Endor",
          "population": "30000000",
          "residents": "",
          "terrain": "forests, mountains, lakes",
        }
      }
      highlighted="highlighted"
      key="Endor"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Endor
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 30000000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: forests, mountains, lakes
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "temperate",
          "name": "Naboo",
          "population": "4500000000",
          "residents": ", , , , , , , , , , ",
          "terrain": "grassy hills, swamps, forests, mountains",
        }
      }
      highlighted="highlighted"
      key="Naboo"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Naboo
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 4500000000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: , , , , , , , , , , 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: grassy hills, swamps, forests, mountains
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "temperate",
          "name": "Coruscant",
          "population": "1000000000000",
          "residents": ", , ",
          "terrain": "cityscape, mountains",
        }
      }
      highlighted="highlighted"
      key="Coruscant"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Coruscant
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 1000000000000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: , , 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: cityscape, mountains
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "temperate",
          "name": "Kamino",
          "population": "1000000000",
          "residents": ", , ",
          "terrain": "ocean",
        }
      }
      highlighted="highlighted"
      key="Kamino"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Kamino
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 1000000000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: , , 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: ocean
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "climate": "temperate, arid",
          "name": "Geonosis",
          "population": "100000000000",
          "residents": "",
          "terrain": "rock, desert, mountain, barren",
        }
      }
      highlighted="highlighted"
      key="Geonosis"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Geonosis
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            climate: temperate, arid
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 100000000000
          </li>
          <li
            key="function now() { [native code] }2"
          >
            residents: 
          </li>
          <li
            key="function now() { [native code] }3"
          >
            terrain: rock, desert, mountain, barren
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          favorite
        </button>
      </div>
    </Card>
  </div>
</Planets>
`;
